# HG changeset patch
# User Mike Miller <mtmiller@octave.org>
# Date 1521078000 25200
#      Wed Mar 14 18:40:00 2018 -0700
# Node ID 120e75853dab19ce34f0b0c9035e142b38424420
# Parent  4229c8ea3227d0dd8aee4586413aa27e4d61d03c
# Parent  d664734ce8eb0ca0c30865489d312cd62a2e27e1
maint: merge stable to default

diff --git a/Makefile.am b/Makefile.am
--- a/Makefile.am
+++ b/Makefile.am
@@ -92,16 +92,17 @@ TAR_OPTIONS = $(REPRODUCIBLE_TAR_FLAGS) 
 
 CONFIG_SUBDIRS = @subdirs@
 
 null =
 ldpreloadsep = ${null}@ldpreloadsep@${null}
 
 image_DATA =
 octdata_DATA =
+octdoc_DATA =
 octetc_DATA =
 octlocale_DATA =
 
 DOC_TARGETS =
 
 BUILT_DISTFILES =
 BUILT_NODISTFILES =
 EXTRA_DIST =
@@ -394,17 +395,17 @@ DIRS_TO_MAKE = \
   $(localveroctfiledir) \
   $(localarchlibdir) \
   $(localapiarchlibdir) \
   $(localverarchlibdir)
 
 installdirs-local:
 	$(MKDIR_P) $(addprefix $(DESTDIR), $(DIRS_TO_MAKE))
 
-install-data-local: installdirs-local doc-interpreter-install-doc-local
+install-data-local: installdirs-local
 
 clean-local: doc-clean
 
 distclean-local:
 	if [ "x${srcdir}" != "x." ]; then rm -f HG-ID; fi
 
 maintainer-clean-local: doc-maintainer-clean
 
diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -209,18 +209,16 @@ OCTAVE_PROG_TAR_REPRODUCIBLE
 ## Programs used when installing Octave.
 AC_PROG_LN_S
 AC_PROG_MKDIR_P
 
 AC_PROG_INSTALL
 INSTALL_SCRIPT="${INSTALL}"
 AC_SUBST(INSTALL_SCRIPT)
 
-OCTAVE_PROG_DESKTOP_FILE_INSTALL
-
 ## Programs used when running Octave.
 OCTAVE_PROG_GHOSTSCRIPT
 OCTAVE_PROG_GNUPLOT
 OCTAVE_PROG_PAGER
 OCTAVE_PROG_PYTHON
 
 ## Programs used to generate icons file formats.
 OCTAVE_PROG_ICOTOOL
diff --git a/doc/interpreter/module.mk b/doc/interpreter/module.mk
--- a/doc/interpreter/module.mk
+++ b/doc/interpreter/module.mk
@@ -187,30 +187,38 @@ HTMLS += %reldir%/octave.html
 
 OCTAVE_HTML_DIR = %reldir%/octave.html
 OCTAVE_HTML_TMP_DIR = $(OCTAVE_HTML_DIR:.html=.htp)
 OCTAVE_HTML_STAMP = $(OCTAVE_HTML_DIR)/.octave-html-stamp
 
 OCTAVE_CSS = %reldir%/octave.css
 HTMLDIR_CSS = $(OCTAVE_HTML_DIR)/octave.css
 
-OCTAVE_QTHELP_STAMP = %reldir%/.octave-qthelp-stamp
 OCTAVE_QTHELP_FILES = %reldir%/octave_interpreter.qhc %reldir%/octave_interpreter.qch
 
+octdoc_DATA += \
+  $(OCTAVE_QTHELP_FILES)
+
 $(srcdir)/%reldir%/octave.info: $(DOC_IMAGES_TXT) $(octave_TEXINFOS)
 %reldir%/octave.dvi: $(DOC_IMAGES_EPS) $(octave_TEXINFOS)
 %reldir%/octave.pdf: $(DOC_IMAGES_PDF) $(octave_TEXINFOS)
 $(OCTAVE_HTML_STAMP): $(DOC_IMAGES_PNG) $(octave_TEXINFOS)
-$(OCTAVE_QTHELP_FILES): $(OCTAVE_QTHELP_STAMP)
-$(OCTAVE_QTHELP_STAMP): $(OCTAVE_HTML_STAMP) $(srcdir)/%reldir%/prepare_qhelp.py
-	$(AM_V_GEN) rm -f $(OCTAVE_QTHELP_FILES) && \
+
+$(OCTAVE_QTHELP_FILES): $(OCTAVE_HTML_STAMP) $(srcdir)/%reldir%/prepare_qhelp.py
+	$(AM_V_GEN)rm -f $(OCTAVE_QTHELP_FILES) && \
 	$(PYTHON) $(srcdir)/%reldir%/prepare_qhelp.py %reldir%/octave_interpreter octave.html && \
 	$(QCOLLECTIONGENERATOR) %reldir%/octave_interpreter.qhcp -o %reldir%/octave_interpreter.qhc >/dev/null && \
 	rm -f %reldir%/octave_interpreter.qhcp %reldir%/octave_interpreter.qhp
 
+## The Qt help collection generator command produces two output files
+## with the same base name: the compressed help (qch) file and the help
+## collection (qhc) file.  Declare the qhc file to depend on the
+## associated qch file, so that the files are built serially.
+%reldir%/octave_interpreter.qhc: %reldir%/octave_interpreter.qch
+
 $(srcdir)/%reldir%/octave.info: %reldir%/octave.texi $(srcdir)/%reldir%/version-octave.texi
 	$(AM_V_MAKEINFO)restore=: && backupdir="$(am__leading_dot)am$$$$" && \
 	am__cwd=`pwd` && $(am__cd) $(srcdir) && \
 	rm -rf $$backupdir && mkdir $$backupdir && \
 	if ($(MAKEINFO) --version) >/dev/null 2>&1; then \
 	  for f in $@ $@-[0-9] $@-[0-9][0-9] $(@:.info=).i[0-9] $(@:.info=).i[0-9][0-9]; do \
 	    if test -f $$f; then mv $$f $$backupdir; restore=mv; else :; fi; \
 	  done; \
@@ -291,20 +299,16 @@ doc_EXTRA_DIST += \
 doc-interpreter-dist-hook:
 	@$(GREP) '#define HAVE_COLAMD 1' $(top_builddir)/config.h > /dev/null || { echo "Documentation creation requires missing COLAMD library.  Cannot package distribution!" ; exit 1; }
 	@$(GREP) '#define HAVE_CHOLMOD 1' $(top_builddir)/config.h > /dev/null || { echo "Documentation creation requires missing CHOLMOD library.  Cannot package distribution!" ; exit 1; }
 	@$(GREP) '#define HAVE_UMFPACK 1' $(top_builddir)/config.h > /dev/null || { echo "Documentation creation requires missing UMFPACK library.  Cannot package distribution!" ; exit 1; }
 	@$(GREP) '#define HAVE_QHULL 1' $(top_builddir)/config.h > /dev/null || { echo "Documentation creation requires missing QHULL library.  Cannot package distribution!" ; exit 1; }
 
 $(MUNGED_TEXI_SRC): $(DOCSTRING_FILES)
 
-doc-interpreter-install-doc-local:
-	$(MKDIR_P) $(octdocdir)
-	cp $(OCTAVE_QTHELP_FILES) $(octdocdir)
-
 ## These two texi files have an additional dependency through the
 ## @EXAMPLEFILE macro.
 %reldir%/oop.texi: $(examples_code_SRC)
 %reldir%/external.texi: $(examples_code_SRC)
 
 %.texi : %.txi %reldir%/munge-texi.pl | %reldir%/$(octave_dirstamp)
 	$(AM_V_GEN)rm -f $@-t $@ && \
 	$(PERL) $(srcdir)/%reldir%/munge-texi.pl $(top_srcdir) $(DOCSTRING_FILES) < $< > $@-t && \
diff --git a/etc/icons/octave.appdata.xml.in b/etc/icons/org.octave.Octave.appdata.xml
rename from etc/icons/octave.appdata.xml.in
rename to etc/icons/org.octave.Octave.appdata.xml
--- a/etc/icons/octave.appdata.xml.in
+++ b/etc/icons/org.octave.Octave.appdata.xml
@@ -14,17 +14,17 @@
   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
   GNU General Public License for more details.
 
   You should have received a copy of the GNU General Public License
   along with Octave; see the file COPYING.  If not, see
   <https://www.gnu.org/licenses/>.
 -->
 <component type="desktop">
-  <id>%OCTAVE_DESKTOP_FILE%</id>
+  <id>org.octave.Octave.desktop</id>
   <metadata_license>FSFAP</metadata_license>
   <project_license>GPL-3.0+</project_license>
   <name>GNU Octave</name>
   <summary>Interactive programming environment for numerical computations</summary>
 
   <description>
     <p>
       GNU Octave is a high-level interpreted language, primarily intended for
diff --git a/etc/icons/octave.desktop.in b/etc/icons/org.octave.Octave.desktop.in
rename from etc/icons/octave.desktop.in
rename to etc/icons/org.octave.Octave.desktop.in
--- a/etc/icons/octave.desktop.in
+++ b/etc/icons/org.octave.Octave.desktop.in
@@ -1,14 +1,14 @@
 [Desktop Entry]
 Name=GNU Octave
 Comment=Scientific Computing
 Comment[de]=Wissenschaftliches Rechnen
 TryExec=%OCTAVE_PREFIX%/bin/octave
-Exec=%OCTAVE_PREFIX%/bin/octave --force-gui %f
+Exec=%OCTAVE_PREFIX%/bin/octave --gui %f
 Icon=octave
 Terminal=false
 Type=Application
 Categories=Education;Science;Math;
 StartupNotify=false
 StartupWMClass=octave-gui
 MimeType=text/x-octave;text/x-matlab;
 Keywords=science;math;matrix;numerical computation;plotting;
diff --git a/etc/module.mk b/etc/module.mk
--- a/etc/module.mk
+++ b/etc/module.mk
@@ -27,16 +27,26 @@ etc_fonts_DATA = \
   %reldir%/fonts/FreeSans.otf \
   %reldir%/fonts/FreeSansBold.otf \
   %reldir%/fonts/FreeSansBoldOblique.otf \
   %reldir%/fonts/FreeSansOblique.otf
 
 %canon_reldir%_EXTRA_DIST += \
   $(etc_fonts_DATA)
 
+appdatadir = $(datadir)/metainfo
+
+appdata_DATA = \
+  %reldir%/icons/org.octave.Octave.appdata.xml
+
+desktopdir = $(datadir)/applications
+
+desktop_DATA = \
+  %reldir%/icons/org.octave.Octave.desktop
+
 icon_IMAGE_FILES = \
   %reldir%/icons/octave-logo.svg \
   %reldir%/icons/octave-sombrero.png
 
 icon_PNG_SIZES = \
   512 \
   256 \
   128 \
@@ -53,40 +63,32 @@ WINDOWS_PNG_ICONS = $(filter %-16.png %-
 
 BUILT_ICONS = \
   $(BUILT_PNG_ICONS) \
   %reldir%/icons/octave-logo.ico
 
 %canon_reldir%_EXTRA_DIST += \
   $(BUILT_ICONS) \
   $(icon_IMAGE_FILES) \
-  %reldir%/icons/octave.appdata.xml.in \
-  %reldir%/icons/octave.desktop.in \
-  %reldir%/icons/octave_branding_samples.svg
+  %reldir%/icons/octave_branding_samples.svg \
+  %reldir%/icons/org.octave.Octave.appdata.xml \
+  %reldir%/icons/org.octave.Octave.desktop.in
 
 image_DATA += \
   $(icon_IMAGE_FILES) \
   %reldir%/icons/octave-logo.ico
 
-VENDOR = www.octave.org
-
 DIRSTAMP_FILES += \
   %reldir%/icons/$(octave_dirstamp)
 
 all-local: all-icons
 
-all-icons: %reldir%/icons/octave.appdata.xml %reldir%/icons/octave.desktop $(BUILT_ICONS)
+all-icons: %reldir%/icons/org.octave.Octave.desktop $(BUILT_ICONS)
 
-%reldir%/icons/octave.appdata.xml: %reldir%/icons/octave.appdata.xml.in | %reldir%/icons/$(octave_dirstamp)
-	$(AM_V_GEN)rm -f $@-t $@ && \
-	$(SED) < $< > $@-t \
-	  -e "s|%OCTAVE_DESKTOP_FILE%|${VENDOR}-octave.desktop|" && \
-	mv $@-t $@
-
-%reldir%/icons/octave.desktop: %reldir%/icons/octave.desktop.in | %reldir%/icons/$(octave_dirstamp)
+%reldir%/icons/org.octave.Octave.desktop: %reldir%/icons/org.octave.Octave.desktop.in | %reldir%/icons/$(octave_dirstamp)
 	$(AM_V_GEN)rm -f $@-t $@ && \
 	$(SED) < $< > $@-t \
 	  -e "s|%OCTAVE_PREFIX%|${prefix}|" && \
 	mv $@-t $@
 
 $(BUILT_PNG_ICONS): %reldir%/icons/octave-logo.svg | %reldir%/icons/$(octave_dirstamp)
 	$(AM_V_GEN)rm -f $@-t $@ && \
 	$(RSVG_CONVERT) -w $(lastword $(subst -, ,$(patsubst %.png,%,$@))) $< > $@-t && \
@@ -97,47 +99,36 @@ all-icons: %reldir%/icons/octave.appdata
 	$(ICOTOOL) --create --raw  $(WINDOWS_PNG_ICONS) > $@-t && \
 	mv $@-t $@
 
 install-data-local: install-icons
 
 uninstall-local: uninstall-icons
 
 install-icons:
-	-if test -n "$(DESKTOP_FILE_INSTALL)"; then \
-	  $(DESKTOP_FILE_INSTALL) --dir=$(DESTDIR)$(datadir)/applications \
-	    --vendor $(VENDOR) %reldir%/icons/octave.desktop; \
-	fi
 	for f in $(BUILT_PNG_ICONS); do \
 	  size=`echo $$f | $(SED) -n -e "s/.*-\([0-9]\+\)\.png/\1/p"`; \
 	  if test -f $$f; then d=; else d="$(srcdir)/"; fi; \
 	  $(MKDIR_P) $(DESTDIR)$(datadir)/icons/hicolor/$${size}x$${size}/apps; \
 	  $(INSTALL_DATA) "$$d$$f" $(DESTDIR)$(datadir)/icons/hicolor/$${size}x$${size}/apps/octave.png; \
 	done
 	$(MKDIR_P) $(DESTDIR)$(datadir)/icons/hicolor/scalable/apps
 	$(INSTALL_DATA) $(srcdir)/%reldir%/icons/octave-logo.svg $(DESTDIR)$(datadir)/icons/hicolor/scalable/apps/octave.svg
-	$(MKDIR_P) $(DESTDIR)$(datadir)/appdata
-	$(INSTALL_DATA) %reldir%/icons/octave.appdata.xml $(DESTDIR)$(datadir)/appdata/$(VENDOR)-octave.appdata.xml
 
 uninstall-icons:
-	if test -n "$(DESKTOP_FILE_INSTALL)"; then \
-	  rm -f $(DESTDIR)$(datadir)/applications/$(VENDOR)-octave.desktop; \
-	fi
 	for f in $(BUILT_PNG_ICONS); do \
 	  size=`echo $$f | $(SED) -n -e "s/.*-\([0-9]\+\)\.png/\1/p"`; \
 	  rm -f $(DESTDIR)$(datadir)/icons/hicolor/$${size}x$${size}/apps/octave.png; \
 	done
 	rm -f $(DESTDIR)$(datadir)/icons/hicolor/scalable/apps/octave.svg
-	rm -f $(DESTDIR)$(datadir)/appdata/$(VENDOR)-octave.appdata.xml
 
 EXTRA_DIST += $(%canon_reldir%_EXTRA_DIST)
 
 %canon_reldir%_CLEANFILES += \
-  %reldir%/icons/octave.appdata.xml \
-  %reldir%/icons/octave.desktop
+  %reldir%/icons/org.octave.Octave.desktop
 
 %canon_reldir%_MAINTAINERCLEANFILES += \
   $(BUILT_ICONS)
 
 CLEANFILES += $(%canon_reldir%_CLEANFILES)
 DISTCLEANFILES += $(%canon_reldir%_DISTCLEANFILES)
 MAINTAINERCLEANFILES += $(%canon_reldir%_MAINTAINERCLEANFILES)
 
diff --git a/m4/acinclude.m4 b/m4/acinclude.m4
--- a/m4/acinclude.m4
+++ b/m4/acinclude.m4
@@ -2836,23 +2836,16 @@ I didn't find bison, or the version of b
 support all the features that are required, but it's only a problem
 if you need to reconstruct parse.cc, which is the case if you're
 building from VCS sources.
 "
     OCTAVE_CONFIGURE_WARNING([warn_bison])
   fi
 ])
 dnl
-dnl Find desktop-file-install program.
-dnl
-AC_DEFUN([OCTAVE_PROG_DESKTOP_FILE_INSTALL], [
-  AC_CHECK_PROG(DESKTOP_FILE_INSTALL, desktop-file-install, desktop-file-install, [])
-  AC_SUBST(DESKTOP_FILE_INSTALL)
-])
-dnl
 dnl Find find program.
 dnl
 ## Prefer GNU find if found.
 AN_MAKEVAR([FIND],  [OCTAVE_PROG_FIND])
 AN_PROGRAM([gfind], [OCTAVE_PROG_FIND])
 AN_PROGRAM([find],  [OCTAVE_PROG_FIND])
 AC_DEFUN([OCTAVE_PROG_FIND], [
   AC_CHECK_PROGS(FIND, [gfind find])
